apps/api/src/historical/historical.service.ts

```diff
@@ -124,12 +141,21 @@ export class HistoricalService {
           .replaceOne({ uuid: doc.uuid }, (last ?? doc) as Player, { upsert: true })
           .lean()
           .exec(),
+        this.historicalLeaderboardService.addHistoricalLeaderboards(
+          time,
+          Player,
+          player,
+          "uuid",
+          true
+        ),
       ]);
     };
 
-    await reset(this.dailyModel, this.lastDayModel, doc);
-    if (isWeekly) await reset(this.weeklyModel, this.lastWeekModel, doc);
-    if (isMonthly) await reset(this.monthlyModel, this.lastMonthModel, doc);
+    await reset(this.dailyModel, this.lastDayModel, doc, HistoricalTimes.DAILY);
```

Comment:
Its probably better to use `Promise.all` and just give it an empty promise if its not weekly or monthly